### PR TITLE
CT-EYB: Group Estimate Your Benefits Section Questions to reduce vertical spacing. #8404

### PR DESCRIPTION
**Pull request by zurbergram**
## Description
https://github.com/department-of-veterans-affairs/va.gov-team/issues/8404#
this is the work for AC 1,2,3 as well as general organization and cleanup work

All work is behind the feature flag`gibctEstimateYourBenefits`

## Acceptance criteria
- [ ] 1. The following text is the first information displayed under the left hand side of the "Estimate Your Benefits" section: "Use the fields below to calculate your benefits". 
- [ ] 2. The "Estimate Your Benefits" section questions are separated into four groups:
    a. Your benefits
    b. About your school
    c. Learning format and schedule
    d. Scholarships and other funding
- [ ] 3. The groupings have the following characteristics:
    a. "Your benefits" section questions are displayed on profile page load.
    b. "About your school", "Learning format and schedule", "Scholarships and other funding" section questions are hidden on profile page load.
    c. When a section questions are displayed all other section questions are hidden.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/gi/components/AccordionItem.jsx [hunk 1]
```diff
@@ -3,72 +3,99 @@ import React from 'react';
...
-      button: props.button,
     };
     this.id = _.uniqueId('accordion-item-');
   }
 
-  toggle() {
-    this.setState({ expanded: !this.state.expanded });
+  componentDidUpdate(prevProps) {
+    if (prevProps.expanded !== this.props.expanded) {
+      // eslint-disable-next-line react/no-did-update-set-state
```
[the remark this answers is not in the document]
Comment (bkjohnson) [reply src/applications/gi/components/AccordionItem.jsx:30]: From [the rule docs](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-did-update-set-state.md):

> Updating the state after a component update will trigger a second `render()` call and can lead to property/layout thrashing.

Are you sure you want to do this?

[the remark this answers is not in the document]
Comment (zurbergram) [reply src/applications/gi/components/AccordionItem.jsx:30]: Yes, unless you have a better solution.

The issue is that the constructor is not being hit again after the `onClick` function is being called and so while the props are being updated the state is not.

See `src/applications/gi/containers/EstimateYourBenefits.jsx`. for what the `onClick` functions are.

There is an AC that states "When a section questions are displayed all other section questions are hidden." and this is my solution to it

File: src/applications/gi/components/search/BenefitsForm.jsx [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+          ]}
+          value={this.props.giBillChapter}
+          alt="Which GI Bill benefit do you want to use?"
+          visible
+          onChange={this.props.eligibilityChange}
+        />
+        {this.props.militaryStatus === 'active duty' &&
+          this.props.giBillChapter === '33' && (
+            <div className="military-status-info warning form-group">
+              <i className="fa fa-warning" />
```
[the remark this answers is not in the document]
Comment (zurbergram) [reply src/applications/gi/components/search/BenefitsForm.jsx]: WIP behind feature flag

File: src/applications/gi/components/search/BenefitsForm.jsx [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+                Post 9/11 GI Bill
+              </a>{' '}
+              recipients serving on Active Duty (or transferee spouses of a
+              service member on active duty) are not eligible to receive a
+              monthly housing allowance.
+            </div>
+          )}
+        {this.props.giBillChapter === '31' && (
+          <div className="military-status-info info form-group">
+            <i className="fa fa-info-circle" />
```
[the remark this answers is not in the document]
Comment (zurbergram) [reply src/applications/gi/components/search/BenefitsForm.jsx]: WIP behind feature flag

File: src/applications/gi/components/profile/BenefitsForm.jsx [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+          ]}
+          value={this.props.giBillChapter}
+          alt="Which GI Bill benefit do you want to use?"
+          visible
+          onChange={this.props.eligibilityChange}
+        />
+        {this.props.militaryStatus === 'active duty' &&
+          this.props.giBillChapter === '33' && (
+            <div className="military-status-info warning form-group">
+              <i className="fa fa-warning" />
```
[the remark this answers is not in the document]
Comment (zurbergram) [reply src/applications/gi/components/profile/BenefitsForm.jsx:103]: WIP behind feature flag

File: src/applications/gi/components/profile/BenefitsForm.jsx [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+                Post 9/11 GI Bill
+              </a>{' '}
+              recipients serving on Active Duty (or transferee spouses of a
+              service member on active duty) are not eligible to receive a
+              monthly housing allowance.
+            </div>
+          )}
+        {this.props.giBillChapter === '31' && (
+          <div className="military-status-info info form-group">
+            <i className="fa fa-info-circle" />
```
[the remark this answers is not in the document]
Comment (zurbergram) [reply src/applications/gi/components/profile/BenefitsForm.jsx:120]: WIP behind feature flag

File: src/applications/gi/components/AccordionItem.jsx [hunk 6]
```diff
@@ -3,72 +3,99 @@ import React from 'react';
...
+    }
+  }
+
+  toggle = () => {
+    const expanded = !this.state.expanded;
+    this.setState({ expanded });
+
+    if (this.props.onClick) {
+      this.props.onClick(expanded);
+    }
```
Comment (bkjohnson) [inline src/applications/gi/components/AccordionItem.jsx:41]: You could probably use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#Optional_chaining_with_function_calls) here like so:

```js
this.props.onClick?.(expanded)
```

File: src/applications/gi/components/AccordionItem.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (zurbergram) [reply src/applications/gi/components/AccordionItem.jsx]: removed usage of `componentDidUpdate`

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Pull request merged**